luaui/Widgets/gui_top_bar.lua

```diff
@@ -2054,22 +3224,125 @@ function shutdown()
 		WG['tooltip'].RemoveTooltip(res .. '_storage')
 		WG['tooltip'].RemoveTooltip(res .. '_current')
 	end
+	
+	if WG['options'] ~= nil then
+		for _, option in ipairs(OPTION_SPECS) do
+			WG['options'].removeOption(getOptionId(option))
+		end
+	end
 end
 
 function widget:Shutdown()
 	--Spring.SendCommands("resbar 1")
+	if widgetHandler:IsWidgetKnown("Top Bar") then
+		widgetHandler:EnableWidget("Top Bar")
+	end
 	shutdown()
 	WG['topbar'] = nil
 end
 
-function widget:GetConfigData()
-	return {
-		autoHideButtons = autoHideButtons
-	}
+function findBPCommand(unitID, unitDefID, cmdList) -- for bp bar only most likely
+	-- TODO: can we get rid of cmdlist? Moving should count as non-idle for at least commanders, and perhaps other units too.
+	local unitExists = false
+	local active = false
+	local builtUnitDefID = nil
+	local mayBeBuilding = false -- will remain false even if builtUnitDefID is non-nil
+	local guardedUnitID = nil -- returned in case we want to check for cycles of idle builders
+
+	local commands = nil
+	-- We only need one command to figure out if the unit is active.
+	-- TODO: Try GetUnitCurrentCommand
+	if UnitDefs[unitDefID].isFactory then
+		commands = Spring.GetFactoryCommands(unitID, 1)
+	else
+		commands = Spring.GetUnitCommands(unitID, -1)
+	end
+
+	if commands then
+		unitExists = true
+		for i = 1, #commands do
+			for _, relevantCMD in ipairs(cmdList) do
+				if commands[i].id == relevantCMD or commands[i].id < 0 then
+					active = true
+					if commands[i].id < 0 and not builtUnitDefID then
+						-- We're building something. A negative ID is the unitdef ID of what's being built.
+						builtUnitDefID = -commands[i].id
+						--Spring.Echo("unit " .. unitID .. " is building unitdef " .. builtUnitDefID)
+					elseif commands[i].id == CMD.GUARD then
+						mayBeBuilding = true -- building can be caused by a guard command
+						guardedUnitID = commands[i].params[1]
+					elseif commands[i].id == CMD.REPAIR then
+						mayBeBuilding = true -- building can be caused by a repair command
+					end
+				end
+			end
+		end
+	end
+
+	return unitExists, active, builtUnitDefID, mayBeBuilding, guardedUnitID
 end
 
-function widget:SetConfigData(data)
-	if data.autoHideButtons ~= nil then
-		autoHideButtons = data.autoHideButtons
+function pid(Kp, Ki, Kd, dt, prevIntegral, prevError, setPoint, measuredValue)
```

Comment:
appears unused

Reply:
thx, for finding this.
Seems to be a left over from Mr. "Major Simplification" ;)

Reply:
fixed version is online